### PR TITLE
Auth

### LINKED ISSUE #55: Create security layer for authentication
Create an authentication / access system, so that the toolkit can be hosted on a server on the internet.

### PR DESCRIPTION
**Pull request by Himyu**
This is still a work in progress but already should have all features for authentication, besides the use of actual user data so atm anyone can log in with whatever password but after login, you get authenticated with JWT and WebSockets can now only connect via API keys.
So please have a first look at it put not merge it yet.

### CODE REVIEW

File: core/app.ts [hunk 1]
```diff
@@ -29,7 +29,9 @@ const main = async (): Promise<void> => {
 
   await moduleService.initialize()
 
-  runServer()
+  lpteService.on('lpt', 'ready', async () => {
```
Comment (Laraakaa) [inline core/app.ts:32]: I think this makes sense. In the future we could maybe refine it so, that the server becomes available but shows in other ways that it's currently not fully ready.

File: core/modules/ModuleService.ts [hunk 2]
```diff
@@ -45,9 +46,59 @@ export class ModuleService {
...
+      })
+    })
+
     const modulePath = this.getModulePath()
     log.debug(`Modules path: ${modulePath}`)
-    const data = await readdirPromise(modulePath)
+
+    this.assets = await this.getAssets()
+
+    // load dri and make sure plugins start loading first
```
Comment (Laraakaa) [inline core/modules/ModuleService.ts]: typo: dir

File: scripts/setup.ts [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  })
+
+  return server.server
+}
+
+const getAuth = async (): Promise<boolean> => {
+  const auth = await inquirer.prompt({
+    type: 'confirm',
+    name: 'enabled',
+    message: 'Do you want to enabled the authentication?',
```
Comment (Laraakaa) [inline scripts/setup.ts]: 'Do you want to enable' (not enabled)

File: scripts/setup.ts [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    type: 'checkbox',
+    name: 'games',
+    choices: [
+      'League of Legends',
+      'Valorant'
+    ]
+  })
+
+  if (choices.games.length <= 0) {
+    console.warn('! Please select at least on game')
```
Comment (Laraakaa) [inline scripts/setup.ts]: on -> one

File: scripts/setup.ts [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+  const assets = await getAll()
+
+  const choices = await inquirer.prompt({
+    type: 'checkbox',
+    name: 'assets',
+    choices: assets.map(a => a.name)
+  })
+
+  if (choices.assets.length <= 0) {
+    console.warn('! Please select at least on Asset')
```
Comment (Laraakaa) [inline scripts/setup.ts]: on -> one

**Review by Laraakaa — APPROVED** [review]

**Review by sdomi4 — APPROVED** [review]
Tested all features, confirmed working with changes to auth.ts to exclude static files.
Remaining changes as discussed.

**Pull request merged**